File: src/main/java/hu/unideb/inf/WebProject/controller/VenueController.java
package hu.unideb.inf.WebProject.controller;

import hu.unideb.inf.WebProject.DTO.VenueDTO;
import hu.unideb.inf.WebProject.service.VenueService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@CrossOrigin(origins = "http://localhost:3000")
@RequestMapping("/venues")
public class VenueController {

    @Autowired
    VenueService venueService;

    @PostMapping("/createVenue")
    ResponseEntity createVenue(@RequestBody VenueDTO venueDTO) {
        return new ResponseEntity(venueService.create(venueDTO), HttpStatus.CREATED);
    }

    @GetMapping("/getAll")
    ResponseEntity getAllVenues() {
        return new ResponseEntity<>(venueService.readAll(), HttpStatus.OK);
    }

    @PatchMapping("/updateVenue")
    ResponseEntity updateVenue(@RequestBody VenueDTO venueDTO) {
        return new ResponseEntity(venueService.update(venueDTO), HttpStatus.OK);
    }

    @DeleteMapping("/deleteVenue")
    ResponseEntity deleteVenue(@RequestBody VenueDTO venueDTO) {
        venueService.delete(venueDTO.venueId());
        return new ResponseEntity(HttpStatus.OK);
    }
}

File: frontend/src/venueController.js
import axios from "axios"

export class VenueController {
    static async createVenue(venueDTO) {
        const respone = await axios.post(`http://localhost:8080/rqimmp/venues/createVenue`, venueDTO)
        return respone.data
    }

    static async getAll() {
        const respons = await axios.get('http://localhost:8080/rqimmp/venues/getAll')
        return respons.data
    }

    static async updateVenue(venueDTO) {
        const respons = await axios.patch('http://localhost:8080/rqimmp/venues/updateVenue', venueDTO)
        return respons.data
    }

    static async deleteVenue(venueDTO) {
        const respons = await axios.delete('http://localhost:8080/rqimmp/venues/deleteVenue' , {data : venueDTO})
        return respons.data
    }
}